lib/src/facts/windows/collection.cc

```diff
@@ -32,13 +33,26 @@ namespace facter { namespace facts {
 
     vector<string> collection::get_external_fact_directories()
     {
-        // Get the user data path
-        TCHAR szPath[MAX_PATH];
-        if (!SUCCEEDED(SHGetFolderPath(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
-            LOG_DEBUG("error finding COMMON_APPDATA: %1%", system_error());
+        if (user::is_admin()) {
+            // Get the common data path
+            TCHAR szPath[MAX_PATH];
+            if (SUCCEEDED(SHGetFolderPath(NULL, CSIDL_COMMON_APPDATA, NULL, 0, szPath))) {
+                path p = path(szPath) / "PuppetLabs" / "facter" / "facts.d";
+                return {p.string()};
+            }
+
+            LOG_WARNING("error finding COMMON_APPDATA, external facts unavailable: %1%", system_error());
+        } else {
+            auto home = user::home_dir();
+            if (!home.empty()) {
+                path p = path(home) / ".puppetlabs" / "opt" / "facter" / "facts.d";
```

Comment:
@peterhuene ah yeah, I agree that's a bit surprising... On the flip side, it is easy to document the per-user paths because it's always `~/.puppetlabs/opt/facter/facts.d` everywhere.